WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java

```diff
@@ -159,12 +164,20 @@ public void showPostActivity(long siteId, long postId) {
         ReaderActivityLauncher.showReaderPostDetail(this, siteId, postId);
     }
 
-    public void showStatsActivityForSite(int localTableSiteId) {
+    public void showStatsActivityForSite(int localTableSiteId, NoteBlockRangeType rangeType) {
         if (isFinishing()) return;
 
-        Intent intent = new Intent(this, StatsActivity.class);
+        Intent intent;
+        if (rangeType == NoteBlockRangeType.FOLLOW) {
+            intent = new Intent(this, StatsViewAllActivity.class);
+            intent.putExtra(StatsAbstractFragment.ARGS_VIEW_TYPE, StatsViewType.FOLLOWERS.ordinal());
```

Comment:
ordinal() always raises a red flag to me. I see that it's already been implemented to work this way. We may want to open a ticket to refactor this somewhere down the line since enums are serializable.

Some good discussion [here](http://stackoverflow.com/questions/3293020/android-how-to-put-an-enum-in-a-bundle).